Userland/Libraries/LibVideo/Color/CodingIndependentCodePoints.h

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2022, Gregory Bertilson <zaggy1024@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/StringView.h>
+
+namespace Video {
+
+enum class ColorPrimaries : u8 {
+    Reserved = 0,    // Should not be used
+    BT709 = 1,       // BT.709 or Rec.709
+    Unspecified = 2, // Used by codecs to indicate that an alternative value may be used
+    BT470M = 4,
+    BT470BG = 5,
+    BT601 = 6, // BT.601 or Rec. 601
+    SMPTE240 = 7,
+    GenericFilm = 8,
+    BT2020 = 9,
+    XYZ = 10,
+    SMPTE431 = 11,
+    SMPTE432 = 12,
+    EBU3213 = 22,
+};
+
+enum class TransferCharacteristics : u8 {
+    Reserved = 0,    // Should not be used
+    BT709 = 1,       // BT.709 or Rec. 709
+    Unspecified = 2, // Used by codecs to indicate that an alternative value may be used
+    BT470M = 4,
+    BT470BG = 5,
+    BT601 = 6, // BT.601 or Rec. 601
+    SMPTE240 = 7,
+    Linear = 8,
+    Log100 = 9,
+    Log100Sqrt10 = 10,
+    IEC61966 = 11,
+    BT1361 = 12,
+    SRGB = 13,
+    BT2020BitDepth10 = 14,
+    BT2020BitDepth12 = 15,
+    SMPTE2084 = 16, // Also known as PQ
+    SMPTE428 = 17,
+    HLG = 18,
+};
+
+enum class MatrixCoefficients : u8 {
+    Identity = 0,    // Applies no transformation to input values
+    BT709 = 1,       // BT.709 or Rec. 709
+    Unspecified = 2, // Used by codecs to indicate that an alternative value may be used
+    Reserved = 3,    // Should not be used
+    FCC = 4,
+    BT470BG = 5, // BT.470 System B or G
+    BT601 = 6,   // BT.601 or Rec. 601
+    SMPTE240 = 7,
+    YCgCo = 8,
+    BT2020NonConstantLuminance = 9,
+    BT2020ConstantLuminance = 10,
+    SMPTE2085 = 11,
+    ChromaticityDerivedNonConstantLuminance = 12,
+    ChromaticityDerivedConstantLuminance = 13,
+    ICtCp = 14,
+};
+
+enum class ColorRange : u8 {
+    Studio = 0, // 16-2
+    Full = 1,
+};
+
+// https://en.wikipedia.org/wiki/Coding-independent_code_points
+struct CodingIndependentCodePoints {
+public:
+    constexpr CodingIndependentCodePoints(ColorPrimaries color_primaries, TransferCharacteristics transfer_characteristics, MatrixCoefficients matrix_coefficients, ColorRange color_range)
+        : m_color_primaries(color_primaries)
+        , m_transfer_characteristics(transfer_characteristics)
+        , m_matrix_coefficients(matrix_coefficients)
+        , m_color_range(color_range)
+    {
+    }
+
+    inline ColorPrimaries color_primaries() const { return m_color_primaries; }
+    inline void set_color_primaries(ColorPrimaries value) { m_color_primaries = value; }
+    inline TransferCharacteristics transfer_characteristics() const { return m_transfer_characteristics; }
```

Comment:
I figured that it would be good to hint that these should be inlined whenever possible since they're simple getters/setters, but I suppose the compiler will do what it wants anyway.

However, now that I think about it, all of these can be constexpr to match the constructor, that way if any more complex code needs to initialize constant CICPs they can. Hopefully that change makes sense?

Reply:
Functions defined inline in a header are implicitly inline

Reply:
Sorry, I don't understand. This is the header and they are declared inline here?

Reply:
Right, sorry for not being more explicit, the keyword `inline` is not required in this context for the compiler to weight inlining these functions very highly. I personally wouldn't try to spray inline all over the library without running it through the profiler and seeing that the functions are showing up.

Reply:
Makes sense.

I'll resolve this since these functions will now be declared constexpr.